protocol/streaming/full_node_streaming_manager.go

```diff
@@ -29,9 +29,12 @@ type FullNodeStreamingManagerImpl struct {
 	ticker *time.Ticker
 	done   chan bool
 
-	// map of clob pair id to stream updates.
-	streamUpdateCache map[uint32][]clobtypes.StreamUpdate
-	numUpdatesInCache uint32
+	// list of stream updates.
+	streamUpdateCache []clobtypes.StreamUpdate
+	// list of subscription ids for each stream update.
+	streamUpdateSubscriptionCache [][]uint32
```

Comment:
nit: cleaner to combine these two since they are used together and assumed to have same length etc

Reply:
added a TODO for now, will address in followup PR